### PR TITLE
Fixes CVE for hoek 6.1.3 by upgrading joi to latest version 17.3.1 in…

### PR DESCRIPTION
**Pull request by seanimam**
## Description

Fixes CVE-2020-36604 for dependency `hoek` 6.3.1 by upgrading joi to latest version 17.3.1 in property-query within PropertyDDS

This required refactoring use of the "Joi" package to the new syntax. The primary changes were the following:

1. The `lazy()` function was replaced with `link()`
   - [Developer docks on link()](https://joi.dev/api/?v=17.13.0#linkref)
   - [Github issue replacing lazy() with link()](https://github.com/hapijs/joi/issues/1968)
 
2. The `valid()` method now takes a list of arguments but not in an array. (e.g. called `.valid("A","B")` versus `.valid(["A","B"])`)
    - [Stack overflow: joi Error: Method no longer accepts array arguments valid](https://stackoverflow.com/questions/58486977/how-to-validate-enum-string-with-joi-latest-version)
    - [Dev docs on valid()](https://joi.dev/api/?v=17.13.0#anyvalidvalues---aliases-equal)
    
3. The `Joi.validate(data, schema, options)` function syntax was replaced with `<Your Joi Schema Object>.validate(data, options)`

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -19106,7 +19106,7 @@ packages:
     engines: {node: ^14.15.0 || ^16.10.0 || >=18.0.0}
     dependencies:
       '@jest/types': 29.6.3
-      '@types/node': 18.19.1
+      '@types/node': 20.12.6
```
Comment (alexvy86) [inline pnpm-lock.yaml:19109]: We shouldn't have all the @types/node updates. Curious, what steps did you follow (e.g. which environment and which folder did you run `pnpm i` from) for this? I just tried the same update to `joi` locally and don't see the @types/node changes...

Comment (seanimam) [reply pnpm-lock.yaml:19109]: I ran `pnpm install joi@^17.13.1` from /experimental/PropertyDDS/packages/property-query

Comment (alexvy86) [reply pnpm-lock.yaml:19109]: Interesting, locally I also get the @types/node changes doing that... can you try manually updating the version in package json and then just running `pnpm i` from the root of the repo?

Comment (tylerbutler) [reply pnpm-lock.yaml]: @seanimam Can you spare a few minutes to update this page in the wiki with anything relevant you learn from this exercise? https://github.com/microsoft/FluidFramework/wiki/Managing-dependencies

I'm trying to make that the "go-to" destination for "how to update deps in the repo" guides.

Comment (alexvy86) [reply pnpm-lock.yaml]: I made a post in our internal Dev channel with my findings, you can use that as a starting point to update the wiki @seanimam

Comment (seanimam) [reply pnpm-lock.yaml]: Just to confirm, I used your method @alexvy86 and the lock file did not have the extraneous changes to node.

File: experimental/PropertyDDS/packages/property-query/CHANGELOG.md [hunk 2]
```diff
@@ -23,6 +23,8 @@
     **Node10 moduleResolution is not supported; it does not support Fluid Framework's API structuring pattern that is used
     to distinguish stable APIs from those that are in development.**
 
+-   `joi` dependency was updated from 14.3.1 to 17.3.1 to address a critical vulnerability exploit [CVE-2020-36604](https://github.com/advisories/GHSA-c429-5p7v-vgjp). This required updating the use of `joi` schema validation function within `property-query` to the new major version syntax.
```
Comment (alexvy86) [inline experimental/PropertyDDS/packages/property-query/CHANGELOG.md:26]: This needs to go into a .changeset file. The release process (for the client release group, specifically) eventually generates release notes and updates individual package changelogs based on the changesets. [This wiki page](https://github.com/microsoft/FluidFramework/wiki/Changesets) should guide you through creating one if necessary.

Comment (seanimam) [reply experimental/PropertyDDS/packages/property-query/CHANGELOG.md]: Updated.

File: .changeset/blood-moon-rising.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@fluid-experimental/property-query": minor
+---
+
+`joi` dependency was updated from 14.3.1 to 17.3.1 to address a critical vulnerability exploit [CVE-2020-36604](https://github.com/advisories/GHSA-c429-5p7v-vgjp). This required updating the use of `joi` schema validation function within `property-query` to the new major version syntax.
```
Comment (alexvy86) [inline .changeset/blood-moon-rising.md:5]: These normally have a change summary line, and then the details, so the release notes can use the summary as title/header entries.
```suggestion
Updated `joi` dependency to latest major version

The `joi` dependency was updated from 14.3.1 to 17.3.1 to address a critical vulnerability exploit [CVE-2020-36604](https://github.com/advisories/GHSA-c429-5p7v-vgjp). This required updating the use of `joi` schema validation function within `property-query` to the new major version syntax.
```

**Review by alexvy86 — APPROVED** [review]
LGTM, let's just update the changeset per the comment below, and wait to see if we can get more reviews.

File: experimental/PropertyDDS/packages/property-query/test/unit/multiple_queries_execution.spec.js [hunk 4]
```diff
@@ -54,7 +54,9 @@ describe("Multiple query execution", () => {
 					],
 				}),
 			)
-				.to.be.eventually.rejectedWith('queryLanguage" must be one of')
+				.to.be.eventually.rejectedWith(
+					'Invalid query 0,queryLanguage: "[0].queryLanguage" must be [queryV1]',
```
Comment (ashishswami22) [inline experimental/PropertyDDS/packages/property-query/test/unit/multiple_queries_execution.spec.js:58]: could you please elaborate why the rejection message has been changed?

Comment (seanimam) [reply experimental/PropertyDDS/packages/property-query/test/unit/multiple_queries_execution.spec.js:58]: The rejection message changed after the major update for joi. I don't know why exactly beyond that.

Comment (seanimam) [reply experimental/PropertyDDS/packages/property-query/test/unit/multiple_queries_execution.spec.js:58]: Pinging for an update @ashishswami22

**Review by tylerbutler — APPROVED** [review]

**Pull request merged**